[10.0] [MIG] catalog_manager

FFernandez-PlanetaTIC opened the pull request:
Hi @bguillot ,

We started a few days ago the migration of the catalog manager from your latest changes.

We are doing TDD so ~~soon~~ in a few months we will have the module working and green tests.

Regards!

OCA#91

cc @PlanetaTIC

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop, connector_prestashbop_catalog_manager: Added
catalog_manager tests of product_image.

MPoch-PlanetaTIC committed:
[FIX]prestashop_catalog_manager: Fix test_export_product_image

MPoch-PlanetaTIC committed:
[ADD+MIG] catalog_manager: Add export tests and pass them for:
- Categories
- Attributes

MPoch-PlanetaTIC committed:
[TMP]connector_prestashop_catalog_manager

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop, connector_prestashop_catalog_manager: added
product_image's test.

MPoch-PlanetaTIC committed:
[FIX]connector_prestasshop_catalog_manager: fix tests

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop, connector_prestashop_catalog_manager: Added
test_export_product_template.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: fix
test_export_product_template.

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop, connector_prestashop_catalog_manager: Added
test_export_product_product.

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop_catalog_manager: Added cassette of
test_export_product_product

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Fix to be able to export product attributes
on product.product exportation.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Fix to export 'filestor'
product image.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Mapped field image_ids of
model product_templates.

MPoch-PlanetaTIC committed:
[ADD]connector_prestashop, connector_prestashop_catalog_manager: Added
delete product images.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Raise exception when trying
to export a product with no prestashop category assigned or with no
product variants.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop, connector_prestashop_catalog_manager: added
fixes of PR 95 and 97 of connector_prestashop in Odoo 9.0

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop, connector_prestashop_catalog_manager: Fixed
flake and pylint errors

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Fixed export_delete_record

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Passing Tests.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: fixed PrestashopProductQuantityListener
usage.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: fixed test_export_stock_qty

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop, connector_prestashop_catalog_manager:defining
_get_invemtory_fields of PrestashopProductQuantityListener as
staticmethod to be called from PrestashopProductProductListener

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Fixed product variants
deletion.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: When removing a
product.attribute from a product.product exported to Prestashop, this
remove will be applied to Prestashop too.
Now, when deleting a product.attribute or a product.attribute.value from
Odoo, it will be deleted to Prestashop too.
deletion.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: When deleting a
product.category exported to Prestashop from Odoo, it will be deleted to
Prestashop too.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: When deleting a
product.template already exported to Prestashop from Odoo, it will be
deleted to Prestashop too.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Fixed error exporting
product.attribute.value twice when assigning a new
product.attribute.value to an already exported product.template.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: flake8 errors fixed.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: passing
test_export_product_product tests.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: passing
test_export_product_product_oncreate test.

LRovira-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Don't duplicate export
variants updating templates

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: When exporting product
template data, do not delete unchanged and unmapped fields on
Prestashop.

FFernandez-PlanetaTIC committed:
[IMP] Set product field "state" to 1 for Prestashop 1.7 compatibility

FFernandez-PlanetaTIC committed:
[FIX] catalog_manager: Fix version and add Marc Poch as contributor

FFernandez-PlanetaTIC committed:
[FIX] catalog_manager: :green_apple: Hack export product image test

FFernandez-PlanetaTIC commented:
Hi @bguillot ,

This is finished and has been working well in some test environment for months.

Could you review it?

Thanks!

FFernandez-PlanetaTIC commented:
@metaminux OCA#93 has been forward-ported in 5499c14

rafaelbn reviewed (approved):
Hi @bguillot could you please review this and merge? :+1:

paulius-sladkevicius commented:
Hey guys, we'll happy to try it and test, but I suppose would be best to merge it and then do a testing on the PR https://github.com/OCA/connector-prestashop/pull/91

paulius-sladkevicius commented:
Sorry, forgot to ping @bguillot who is mainly can do that :-)

bguillot commented on connector_prestashop/__manifest__.py:
```diff
@@ -15,7 +15,7 @@
         "product_multi_category",  # oca/product-attribute
         "connector_ecommerce",  # oca/connector-ecommerce
-        "base_multi_image",  # oca/product-attribute
+        "product_multi_image",
```
I think it's not useful to depend on product_multi_image in connector_prestashop module, it is needed mainly for export.

FFernandez-PlanetaTIC replied:
Hi @bguillot,

Thanks you for your review.

`product_multi_image` allows to show imported images in product sheet, `base_multi_image` does not.

I think it makes no sense to import images if we don't show them.

Regards

bguillot commented on connector_prestashop_catalog_manager/models/product_template/exporter.py:
```diff
@@ -242,53 +244,25 @@ def _get_product_category(self, record):
-                'product_features': {
-                    'product_feature': self._get_template_feature(record)},
-                'accessories': {
-                    'accessory': self._get_product_links(record)},
```
Why do you remove the export of product links and attributes ?

FFernandez-PlanetaTIC replied:
@bguillot ,

About the export of product links, it requires module `product_template_multi_link`, that's why we removed it.

I'm sorry, it wasn't the smartest way.

Anyway, I think this should be done in another module to not add more dependencies to catalog manager.

We have done so in OCA/connector-prestashop#123 and ~~OCA/connector-prestashop#132~~ OCA/connector-prestashop#126 .

What do you think?

cc @MPoch-PlanetaTIC

FFernandez-PlanetaTIC replied:
About the export of product attributes, I think "apples and oranges were mixing", I try to explain:

These are combinations and their option values:

![image](https://user-images.githubusercontent.com/39995504/65897502-afd0f600-e3af-11e9-832a-652837ebb52b.png)

And these are features and their values:

![image](https://user-images.githubusercontent.com/39995504/65897553-cd9e5b00-e3af-11e9-95ef-3d492bc91658.png)

Removed code was trying to save option values into product features:

https://github.com/akretion/connector-prestashop/pull/8/files#diff-da8e6cee6ea3159817655a0fef709826L245

Or that is what it seems to me :dizzy_face:

PD: OCA/connector-prestashop#126 imports PS features into Odoo `product_custom_info`

bguillot reviewed (commented):
Thanks you this work ! 

Can you answer my two questions before we can merge ?

FFernandez-PlanetaTIC committed:
[IMP] catalog_manager: New PrestaShop fields for low stock

paulius-sladkevicius commented:
Hey @bguillot sorry for pushing, however wondering if there is anything to do by @FFernandez-PlanetaTIC and it can be merged already. Thanks!

MPoch-PlanetaTIC committed:
[FIX] connector_catalog_manager: Apply direct mapping on default_code.

MPoch-PlanetaTIC committed:
[FIX] connector_prestashop_catalog_manager: export translations
When editing or creating a translation, force a write on source model
instance to export its translation.

paulius-sladkevicius commented on connector_prestashop_catalog_manager/models/product_template/common.py:
```diff
@@ -54,23 +55,22 @@ class PrestashopProductTemplateListener(Component):
     def on_record_write(self, record, fields=None):
         """ Called when a record is written """
         record.with_delay().export_record(fields=fields)
         if 'minimal_quantity' in fields:
```
While changing `minimal_quantity` it brings recursion error:

```
File "/home/odoo/projec/addons_external/connector/producer.py", line 37, in write
  result = super(Base, self).write(vals)
File "/home/odoo/projec/addons_external/component_event/models/base.py", line 101, in write
  result = super(Base, self).write(vals)
File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
  self._write(old_vals)
File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3704, in _write
  field.write(self.with_context(rel_context), vals[name])
File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/fields.py", line 2228, in write
  comodel.browse(act[1]).write(act[2])
File "/home/odoo/projec/addons_external/connector/producer.py", line 37, in write
  result = super(Base, self).write(vals)
File "/home/odoo/projec/addons_external/component_event/models/base.py", line 104, in write
  self._event('on_record_write').notify(record, fields=fields)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 190, in notify
  event(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/connector_prestashop_catalog_manager/models/product_template/common.py", line 72, in on_record_write
  'minimal_quantity': binding.minimal_quantity
File "/home/odoo/projec/addons_external/connector/producer.py", line 37, in write
  result = super(Base, self).write(vals)
File "/home/odoo/projec/addons_external/component_event/models/base.py", line 104, in write
  self._event('on_record_write').notify(record, fields=fields)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 190, in notify
  event(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/connector_prestashop_catalog_manager/models/product_template/common.py", line 72, in on_record_write
  'minimal_quantity': binding.minimal_quantity
File "/home/odoo/projec/addons_external/connector/producer.py", line 37, in write
  result = super(Base, self).write(vals)
File "/home/odoo/projec/addons_external/component_event/models/base.py", line 104, in write
  self._event('on_record_write').notify(record, fields=fields)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 190, in notify
  event(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/component_event/components/event.py", line 160, in func_wrapper
  return func(*args, **kwargs)
File "/home/odoo/projec/addons_external/connector_prestashop_catalog_manager/models/product_template/common.py", line 72, in on_record_write
....
```

FFernandez-PlanetaTIC replied:
It has been fixed, thanks @paulius-sladkevicius !

paulius-sladkevicius replied:
@FFernandez-PlanetaTIC thanks. We do heavy testing now, so happy to contribute at least with issue.

FFernandez-PlanetaTIC committed:
[FIX] catalog_manager: Do not export position_in_category
to avoid PrestaShop issues

FFernandez-PlanetaTIC committed:
[FIX] catalog_manager: Fix recursion error on change minimal_quantity

FFernandez-PlanetaTIC commented:
Hi @paulius-sladkevicius ,

Although we have made minor corrections during the last days , the module is stable and it is running in some production environments.

Thank you for your review!

MPoch-PlanetaTIC committed:
[FIX]connector_catalog_manager: Fixed error when writing on multiple
instances of ir_translation at once.

MPoch-PlanetaTIC committed:
[FIX] catalog_manager: Fixed Travis error.

paulius-sladkevicius commented:
Hi @FFernandez-PlanetaTIC 

I'm testing it further. Now I got recursion looks on translations. To reproduce, change any configuration on the e.g. Sales Configuration. Looks it does write on the `res.company` and then stuck with recursion. Thanks!

```
Traceback (most recent call last):
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/odoo/addons/web/controllers/main.py", line 892, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/odoo/addons/web/controllers/main.py", line 884, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/odoo/project/addons_external/connector/producer.py", line 30, in create
    record = super(Base, self).create(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 95, in create
    record = super(Base, self).create(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3855, in create
    self._fields[key].determine_inverse(record)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/fields.py", line 1048, in determine_inverse
    self.inverse(records)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/fields.py", line 634, in _inverse_related
    other[field.name] = record_value[record]
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 5241, in __setitem__
    return self._fields[key].__set__(self, value)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/fields.py", line 954, in __set__
    record.write({self.name: write_value})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/res/res_company.py", line 350, in write
    return super(Company, self).write(values)
  File "/home/odoo/project/addons_external/connector/producer.py", line 37, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 101, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
    self._write(old_vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3677, in _write
    self.env['ir.translation']._sync_terms_translations(self._fields[name], self)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/ir/ir_translation.py", line 503, in _sync_terms_translations
    outdated.write({'state': 'to_translate'})
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 31, in write
    self.write_on_source_model()
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 24, in write_on_source_model
    instance.write({fieldname: untranslated_content})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/res/res_company.py", line 350, in write
    return super(Company, self).write(values)
  File "/home/odoo/project/addons_external/connector/producer.py", line 37, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 101, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
    self._write(old_vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3677, in _write
    self.env['ir.translation']._sync_terms_translations(self._fields[name], self)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/ir/ir_translation.py", line 503, in _sync_terms_translations
    outdated.write({'state': 'to_translate'})
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 31, in write
    self.write_on_source_model()
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 24, in write_on_source_model
    instance.write({fieldname: untranslated_content})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/res/res_company.py", line 350, in write
    return super(Company, self).write(values)
  File "/home/odoo/project/addons_external/connector/producer.py", line 37, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 101, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
    self._write(old_vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3677, in _write
    self.env['ir.translation']._sync_terms_translations(self._fields[name], self)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/ir/ir_translation.py", line 503, in _sync_terms_translations
    outdated.write({'state': 'to_translate'})
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 31, in write
    self.write_on_source_model()
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 24, in write_on_source_model
    instance.write({fieldname: untranslated_content})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/res/res_company.py", line 350, in write
    return super(Company, self).write(values)
  File "/home/odoo/project/addons_external/connector/producer.py", line 37, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 101, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
    self._write(old_vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3677, in _write
    self.env['ir.translation']._sync_terms_translations(self._fields[name], self)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/ir/ir_translation.py", line 503, in _sync_terms_translations
    outdated.write({'state': 'to_translate'})
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 31, in write
    self.write_on_source_model()
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 24, in write_on_source_model
    instance.write({fieldname: untranslated_content})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/res/res_company.py", line 350, in write
    return super(Company, self).write(values)
  File "/home/odoo/project/addons_external/connector/producer.py", line 37, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/project/addons_external/component_event/models/base.py", line 101, in write
    result = super(Base, self).write(vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3568, in write
    self._write(old_vals)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/models.py", line 3677, in _write
    self.env['ir.translation']._sync_terms_translations(self._fields[name], self)
  File "/home/odoo/.venv-odoo/lib/python2.7/site-packages/odoo/addons/base/ir/ir_translation.py", line 503, in _sync_terms_translations
    outdated.write({'state': 'to_translate'})
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 31, in write
    self.write_on_source_model()
  File "/home/odoo/project/addons_external/connector_prestashop_catalog_manager/models/ir_translation.py", line 24, in write_on_source_model
    instance.write({fieldname: untranslated_content})
  File "/home/odoo/odoo/addons/account/models/company.py", line 119, in write
    return super(ResCompany, self).write(values)
  File "/home/odoo/odoo/addons/product/models/res_company.py", line 55, in write
    return super(ResCompany, self).write(values)
``

paulius-sladkevicius commented:
Please ignore my latest issue, with latest commits I can't reproduce it.

FFernandez-PlanetaTIC committed:
[FIX] catalog_manager: Fix export of variants stock from product

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: importing product brands from Prestashop.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Fixed class name of
ProductCategoryDeleter

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Adding view for product brand binding.
Also set _export_node_name variable for product brand exportation.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Export product.brand.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: When exporting a
product.brand to prestashop exporting as actived.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Persmissions on prestashop.product.brand
model

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Added tests for
prestashop.product.brand.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: fixed travis errors.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Added tests for product.brand import.

MPoch-PlanetaTIC committed:
[FIX] connector_prestashop_catalog_manager: set a direct mapping on
field product_brand_id of product.template model, to export brand's
product on prestashop.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop_catalog_manager: Tests fixed

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Fixed yaml files to point localhost.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: Before import product.brand check if there is
a product.brand to be imported.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop: self.prestashop_record.get('id_manufacturer')
returns a string value, and importation product.brand failed.

MPoch-PlanetaTIC committed:
[FIX]connector_prestashop, connector_prestashop_catalog_manager: Mapped
Prestashop field "visibility"

MPoch-PlanetaTIC committed:
[FIX] connector_prestashop, connector_prestashop_catalog_manager:
In Prestashop images can be assigned to a specific set of combinations,
this fix preserve this image assignation when importing/exporting
products from/to Prestashop.

MPoch-PlanetaTIC committed:
[FIX] connector_prestasshop_catalog_manager: When changing impact_price
in a product variant, export it to Prestashop.

MPoch-PlanetaTIC committed:
[FIX] connector_prestashop_catalog_manager: Fix when exporting a product
with a single option_value to Prestashop
